### ISSUE TITLE
Does not support portrait aspect ratio videos

### ISSUE THREAD

**Issue by StevenJLawson** [protyposis/ITEC-MediaPlayer#6]
Try opening a video that is 720x1280 for example.

**Comment by protyposis** [comment]
Hi, videos encoded in "real" portrait resolution (720x1280, 1080x1920) are displayed correctly. What doesn't work, are videos encoded in landscape resolution with the rotation encoded in the metadata (e.g. 1920x1080 + 90 degrees rotation = 1080x1920).

While the MediaFormat contains the rotation field ("rotation-degrees"), the API does not expose a public key to read it... which is because the MediaCodec seems to be supposed to handle the rotation internally when decoding, but this seems not to be working.

Long story short: I cannot find a non-hacky way to determine the rotation of a video and display it correctly. Hopefully somebody else can help on this?

**Comment by protyposis** [comment]
Fixed in 60ef81e / v2.2.2

From API21, rotated videos are correctly rendered by the decoder and the video dimensions are reported with the rotation applied. API20 and below do not automatically rotate videos, do not expose the rotation value, and the physical dimensions are reported.

Rotated video is therefore only correctly supported from API21 (due to lack of support in the Android system).

More information on this topic can be found in [this ExoPlayer issue](https://github.com/google/ExoPlayer/issues/91).
